src/client/api/metrics.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * (C) Copyright 2020-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/utsname.h>
+#include <daos/common.h>
+#include <daos/job.h>
+#include <daos/tls.h>
+#include <gurt/telemetry_common.h>
+#include <gurt/telemetry_consumer.h>
+#include <gurt/telemetry_producer.h>
+
+#define INIT_JOB_NUM		1024
+bool client_metric;
+bool client_metric_retain;
+
+#define MAX_IDS_SIZE(num) (num * D_TM_METRIC_SIZE)
+/* The client side metrics structure looks like
+ * root/job_id/pid/....
+ */
+int
+dc_tm_init(void)
+{
+	struct d_tm_node_t *job_node;
+	struct d_tm_context *current_ctx;
+	struct daos_thread_local_storage *dtls;
+	int		    metrics_tag;
+	pid_t		    pid;
+	int		    rc;
+
+	d_getenv_bool(DAOS_CLIENT_METRICS_ENV, &client_metric);
```

Comment:
would it be useful if this was controlled by the agent or the server instead of an environment variable?

Reply:
Indeed, though I believe @mjmac  will move the client metrics root to agent  based on this PR.

Reply:
> would it be useful if this was controlled by the agent or the server instead of an environment variable?

Hmm. This is an interesting idea. Certainly we could add some control via the GetAttachInfo upcall so that client applications have telemetry enabled/disabled according to an agent or server config setting. I think we can add that later though, as another option in addition to this environment variable.

Reply:
For the record, I went ahead and implemented this in my follow-on PR: https://github.com/daos-stack/daos/pull/13545